sql-plugin/src/main/scala/com/nvidia/spark/rapids/NvtxWithMetrics.scala

```diff
@@ -18,6 +18,15 @@ package com.nvidia.spark.rapids
 
 import ai.rapids.cudf.{NvtxColor, NvtxRange}
 
+object NvtxWithMetrics {
+  def apply(name: String, color: NvtxColor, metric: Option[GpuMetric] = None): NvtxRange = {
```

Comment:
Curious why this has a default argument?  The caller should be using a regular `NvtxRange` if they don't have a possible metric to pass as a parameter.

Reply:
This was added in response to https://github.com/NVIDIA/spark-rapids/pull/2823#discussion_r659802704

Reply:
I see what you mean ... I will remove the default.